sql/core/src/test/scala/org/apache/spark/sql/DataFrameFunctionsSuite.scala

```diff
@@ -1045,6 +1045,31 @@ class DataFrameFunctionsSuite extends QueryTest with SharedSQLContext {
       Seq(Row(null), Row(null))
     )
 
+    checkAnswer(
+      df.selectExpr("array_position(array(1), 1.23D)"),
+      Seq(Row(0L), Row(0L))
+    )
+
+    checkAnswer(
+      df.selectExpr("array_position(array(1), 1.0D)"),
+      Seq(Row(1L), Row(1L))
+    )
+
+    checkAnswer(
+      df.selectExpr("array_position(array(1.23D), 1)"),
+      Seq(Row(0L), Row(0L))
+    )
```

Comment:
What about `array_position(array(1.0D), 1)`?

Reply:
@ueshin Sure.. i will add this case.